processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -806,7 +781,7 @@ public JavaExpressionMappingBuilder javaExpression(String javaExpression) {
         }
 
         public PropertyMapping build() {
-            Assignment assignment = new SourceRHS( javaExpression, null, existingVariableNames );
+            Assignment assignment = new SourceRHS( javaExpression, null, existingVariableNames, "" );
```

Comment:
The only place were this message is used is in `MappingResolver`. This is not called in an expression since there is nothing to resolve. So there is never an error message to print. I guess this comes from generalizing stuff.

To avoid this, I could make an overloaded constructor to `sourceRHS`.. WDYT?